src/client/client.ts

```diff
@@ -1,15 +1,13 @@
 import {IsImplementedCache} from './../helper/isImplementedCache';
-import request from 'request-promise';
 import {Guid} from 'guid-typescript';
 import {BaseClient} from './baseClient';
 import {Config, defaultConfig} from './clientConfig';
 import {Tooling, Fetch, GetCsdsEntry} from '../types/tooling';
 import {CsdsClient} from '../helper/csdsClient';
-import {RequestError} from 'request-promise/errors';
 import {RETRIABLE_ERRORS} from '../helper/networkErrors';
-import {sleep} from '../helper/common';
 import {Response} from '../types/response';
-import {FetchOptions} from '../types/fetchOptions';
+import {RequestError, RetryObject} from 'got';
```

Comment:
It's optional, but we could use `import type` here, since we are just importing types here, right?